# **Estimation And Confidence Intervals:**

a. Build 99% Confidence Interval Using Sample Standard Deviation

In [1]:
from scipy import stats
import pandas as pd
import numpy as np

By assuming the given sample as the population,
data=[1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]

In [2]:
data=pd.Series([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

In [3]:
data.mean()

1.2386666666666666

In [4]:
data.std()

0.19316412956959936

To find the confidence interval for the mean with unknown standard deviation,we have to consider the sample standard deviation(s) as the population standard deviation(Sigma).
    
    t=¯x−μs√n
    t=¯x+μs√n
In python    

    stats.t.interval(CL,n-1,mean(),std())

For 99% confidence interval,

    we got the mean=1.2387
    we got the std(s)=0.19316
    we got the sample size n=15

In [5]:
data_ci_sd=stats.t.interval(0.99,14,1.238667,0.193164)
print( 'The 99% confidence interval using sample standard deviation is:', np.round(data_ci_sd, 4))

The 99% confidence interval using sample standard deviation is: [0.6636 1.8137]


**T-Distribution:**

t-distibution is a type of normal distribution,we use If the sample size is small and the population standard deviation is unknown.Whether it is normal or t- distribution,it is used to describe the observations where observations will be very close to the population mean value.

Here we're using the t-distribtion only because we don't know the population standard deviation and the sample size is also small.
So,we will consider sample standard deviation as population standard deviation in t-distribution.

 b.Build 99% Confidence Interval Using Known Population Standard Deviation

In [ ]:
data=pd.Series([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

In [6]:
data.describe()

count    15.000000
mean      1.238667
std       0.193164
min       0.850000
25%       1.155000
50%       1.250000
75%       1.345000
max       1.550000
dtype: float64

In [7]:
data_ci = stats.norm.interval(0.99,loc = data.mean(),scale =0.2)
print( 'The 99% confidence interval using population standard deviation is:', np.round(data_ci, 4))

The 99% confidence interval using population standard deviation is: [0.7235 1.7538]
